### DISCUSSION TITLE
on:click not working with Button component

### DISCUSSION THREAD

**Discussion by HonigBaer1** [huntabyte/shadcn-svelte#1571] [Help] [answered]
Hello,

I am using SvelteKit with Svelte 5 and the latest version of Shadcn-Svelte. I have a problem with using the on:click feature of the <Button/> component. When I try to add an on:click event to the button, the function never runs. I looked throught he repo to find examples on how to use it, but nothing that helped me. Below I have my example which doesnt work:

I installed the component using the npx command from the docs:
`npx shadcn-svelte@next add button`

+page.svelte
```svelte
<script lang="ts">
	import { Button } from '$lib/components/ui/button';
</script>

<Button on:click={() => alert("Hello World")}>Click me</Button>
```

If there is any more information I can provide, please let me know.

**Comment by shyakadavis** [comment]
```diff
<script lang="ts">
	import { Button } from '$lib/components/ui/button';
</script>

--<Button on:click={() => alert("Hello World")}>Click me</Button>
++<Button onclick={() => alert("Hello World")}>Click me</Button>
```

i.e. `on:click` -> `onclick`

**Comment by lightningorb** [comment]
Why was on:click renamed to onclick? This seems to break user expectations? What about modifiers like once | preventDefault etc. ? Finally could this get documented?

**Reply by shyakadavis** [reply, marked as answer]
This is a Svelte 5 thing: https://svelte.dev/docs/svelte/v5-migration-guide#Event-changes

> Event handlers have been given a facelift in Svelte 5. Whereas in Svelte 4 we use the on: directive to attach an event listener to an element, in Svelte 5 they are properties like any other (in other words - remove the colon):

**Reply by lightningorb** [reply]
Duh. :)

Thanks so much for your response.
